### DISCUSSION TITLE
How do I open nvim-tree in startup without overiding the NvChad nvim-tree setup options.

### DISCUSSION THREAD

**Discussion by FosRexx** [NvChad/NvChad#2893] [Q&A] [answered]
I want to open nvim-tree in startup and following [this](https://github.com/nvim-tree/nvim-tree.lua/wiki/Open-At-Startup#always-open-nvim-tree) and [this](https://nvchad.com/docs/config/plugins), I made this lua file in `.config/nvim/lua/plugins/nvim-tree.lua`:
```
return {
  "nvim-tree/nvim-tree.lua",
  lazy = false,
  config = function ()
    require("nvim-tree").setup {}
    require("nvim-tree.api").tree.open()
  end
}
```

But doing that overrides the default "look"/"options" of nvim-tree, the one provided by NvChad. I am aware that this is caused by the line `require("nvim-tree").setup {}` but if I do not add that line then nvim-tree complains saying `[NvimTree] nvim-tree setup not called` and then nvim-tree wont open with error `E492: Not an editor command: NvimTreeToggle`.

And so I was just wondering if their is a way to open nvim-tree on startup while keeping the NvChad's setup.

**Comment by siduck** [comment]
@FosRexx you can run the open() in vim.schedule or vim.defer_fn

**Reply by FosRexx** [reply]
I am sorry but I am very new to NeoVim and so i dont know where to add that command as in which file do I add that command?

**Reply by FosRexx** [reply]
I added `require("nvim-tree.api").tree.open()` after `require("mappings")` inside the `vim.schedule()` function in `.config/nvim/init.lua` and that seems to have worked.

**Comment by FosRexx** [comment, marked as answer]
After learning more about neovim I did this quite differently than the original answer, I wont explain the reasons much here be I will just share what I have done.

First i added this line at the end of `~/.config/nvim/init.lua` file:
```
require "custom"
```

Then I created a `custom` directory in `~/.config/nvim/lua/`, now we have a directory like `~/.config/nvim/lua/custom/`

Then I created a `init.lua` file inside `~/.config/nvim/lua/custom/`. Now in `~/.config/nvim/lua/custom/init.lua`, I added this line:
```
require("custom.nvimtree")
```

Now I again created a new file named `nvimtree.lua` inside `~/.config/nvim/lua/custom/`. Now in `~/.config/nvim/lua/custom/nvimtree.lua`, I added this which is copied directly from the [nvim-tree wiki](https://github.com/nvim-tree/nvim-tree.lua/wiki/Open-At-Startup#always-open-nvim-tree):
```
-- Open nvim-tree on startup
local function open_nvim_tree()
  -- open the tree
  require("nvim-tree.api").tree.open()
end

vim.api.nvim_create_autocmd({ "VimEnter" }, { callback = open_nvim_tree })
```

This setup works and nvim-tree opens on startup.
